uspace/app/bdsh/input.c

```diff
@@ -224,6 +281,21 @@ errno_t process_input(cliuser_t *usr)
 	return rc;
 }
 
+errno_t process_input(cliuser_t *usr)
+{
+	list_t alias_hups;
+	list_initialize(&alias_hups);
+
+	errno_t rc = process_input_nohup(usr, &alias_hups);
+
+	list_foreach_safe(alias_hups, cur_link, next_link) {
```

Comment:
If you do the processing as I recommended above, you won't need this loop since the list should be empty on return. Putting an `assert()` here checking that it actually will be empty might also help.